New "Already own a domain" component in the domain suggestions results

hambai opened the pull request:
Add new component to show the "Already own a domain" tip at the bottom of the domain suggestions according to our domains redesign

#### Changes proposed in this Pull Request

* This is a new component that'll render on the domain suggestions page below the suggestion results replacing the old card that we render within the suggestion results

Here's how it looks:

Before:
<img width="1089" alt="Screenshot 2021-08-20 at 13 50 07" src="https://user-images.githubusercontent.com/1355045/130222767-296c2634-db16-4c65-9e57-64bf231eb7ad.png">

After:
<img width="1086" alt="Screenshot 2021-08-20 at 13 49 40" src="https://user-images.githubusercontent.com/1355045/130222759-64818c3c-d0f9-4df6-8c3a-ab6fdfc324ee.png">


#### Testing instructions

* Open up the branch and go to a site and add a domain - you should not see the old row at the bottom of the domain suggestions results and you should see the new tip at the bottom after the results.

hambai committed:
Add new component to show the "Already own a domain" tip at the bottom of the domain suggestions according to our domains redesign

hambai committed:
* fix the css to render properly on mobile

hambai committed:
* fix the component top margin

poligilad-auto commented:
Looking good @hambai, only thing I noticed is on mobile the icon should be aligned to the top.

![image](https://user-images.githubusercontent.com/83961704/130405513-8ed283ab-c583-42ae-af38-dd4eafeb3af0.png)

hambai committed:
fix e2e tests

hambai committed:
* fix mobile style for the notice
* e2e tests again

hambai committed:
* fix the color of the icon and the text

worldomonation commented:
Updated selector in `findDomainComponent` looks correct, but both desktop and mobile e2e tests fail at this step.

hambai committed:
* Fix the e2e tests

TIL: `display:contents` is buggy and not recommended - see https://developer.mozilla.org/en-US/docs/Web/CSS/display#box

hambai commented:
> Updated selector in `findDomainComponent` looks correct, but both desktop and mobile e2e tests fail at this step.

Hey, finally figured it out - a button with `display:contents` CSS just doesn't work with CSS selectors and that's what I was trying to use to make the button appear as a normal link. I believe this is related to this:

> These elements don't produce a specific box by themselves. They are replaced by their pseudo-box and their child boxes. Please note that the CSS Display Level 3 spec defines how the contents value should affect "unusual elements" — elements that aren’t rendered purely by CSS box concepts such as replaced elements. See Appendix B: Effects of display: contents on Unusual Elements for more details.
> Due to a bug in browsers this will currently remove the element from the accessibility tree — screen readers will not look at what's inside. See the Accessibility concerns section below for more details.



see: https://developer.mozilla.org/en-US/docs/Web/CSS/display#box

delputnam reviewed (approved):
This LGTM. Nice job.

worldomonation reviewed (approved):
Approved from Quality Squad as well. Thanks for fixing the e2e tests!

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/6470455

Thank you @hambai for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/6470455).
